src/test/java/testsuite/integration/container/AuroraMysqlFailoverIntegrationTest.java

```diff
@@ -434,13 +435,56 @@ public void test_failoverTimeoutMs() throws SQLException, IOException {
       long invokeStartTimeMs = System.currentTimeMillis();
       SQLException e = assertThrows(SQLException.class, () -> stmt.executeQuery("SELECT 1"));
       long invokeEndTimeMs = System.currentTimeMillis();
-      
+
       assertEquals(MysqlErrorNumbers.SQL_STATE_UNABLE_TO_CONNECT_TO_DATASOURCE, e.getSQLState());
       long duration = invokeEndTimeMs - invokeStartTimeMs;
       assertTrue(duration < 15000); // Add in 5 seconds to account for time to detect the failure
     }
   }
 
+  @Test
+  public void test_keepAutoCommitParameter()
```

Comment:
I'd split this test into two simple tests. One checks behaviour without keepSessionStateOnFailover setting, one checks logic with keepSessionStateOnFailover.